contrib/ansible/deploy.yaml

```diff
@@ -0,0 +1,62 @@
+---
+- hosts: xsnippet
+  tasks:
+    - name: ensure python 2 is installed
+      raw: test -e /usr/bin/python || (apt -y update && apt install -y python-minimal)
+  gather_facts: false
+
+- hosts: xsnippet
+  vars:
+    xsnippet_image: xsnippet/xsnippet-api:latest
+  tasks:
+    - name: update apt cache
+      apt:  update_cache=yes cache_valid_time=3600
+
+    - name: install essential packages
+      apt: "name={{ item }} state=latest"
+      with_items:
+        - apt-transport-https
+        - ca-certificates
+        - curl
+        - software-properties-common
+
+    - name: add Docker gpg key
+      shell: "curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -"
+
+    - name: add Docker packages repo
+      apt_repository:
+          repo: "deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+          state: present
+
+    - name: update apt cache
+      apt: update_cache=yes
+
+    - name: install Docker
+      apt: name=docker-ce state=latest
+
+    - name: initialize Docker Swarm
+      shell: "docker swarm init --advertise-addr `ip -o a | grep eth | head -n1 | awk '{print $4}' | cut -d '/' -f 1` || docker stack ls"
```

Comment:
Jesus.. what does this line return?

```
`ip -o a | grep eth | head -n1 | awk '{print $4}' | cut -d '/' -f 1`
```

Can we use ansible facts instead to retrieve proper IP address?

Reply:
Will do! (the first IPv4 address on the interfaces starting with `eth` :P )